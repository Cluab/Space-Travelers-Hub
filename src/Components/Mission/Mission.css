.mission-container {
  border: 1px solid rgba(0, 0, 0, 0.336);
  border-top: none;
}

.mission-info {
  margin-top: 0;
  padding-left: 0;
  list-style: none;
  display: flex;
}

.mission-info :nth-child(1) {
  font-weight: 500;
  padding: 10px;
  width: 15%;
  border-right: 1px solid rgba(0, 0, 0, 0.336);
}

.mission-info :nth-child(2) {
  font-size: 15px;
  padding: 10px;
  width: 65%;
  border-right: 1px solid rgba(0, 0, 0, 0.336);
}

.mission-info :nth-child(3) {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 15%;
  padding: 10px;
  border-right: 1px solid rgba(0, 0, 0, 0.336);
}

.mission-info :nth-child(3) span {
  border-right: none;
  width: fit-content;
  font-size: 13px;
  padding: 5px;
  border-radius: 8px;
  color: white;
}

.mission-status-Active {
  background-color: #007bff;
}

.mission-status {
  background-color: rgb(150, 148, 148);
}

.mission-info :nth-child(4) {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  width: 15%;
}

.mission-info :nth-child(4) button {
  border-right: 2px solid;
  height: 30px;
  width: 100%;
  padding: 0;
  background-color: white;
  border-radius: 5px;
}

.joinMission {
  color: rgb(88, 87, 87);
  border: 2px solid rgb(88, 87, 87);
}

.leaveMission {
  color: red;
  border: 2px solid red;
}
