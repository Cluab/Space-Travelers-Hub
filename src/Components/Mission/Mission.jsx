import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission } from '../../Redux/Missions/mission';
import './Mission.css';

const Mission = ({
  id, name, description, member,
}) => {
  const dispatch = useDispatch();
  return (
    <div className="mission-container">
      <div className="mission-info">
        <div data-testid="test-name">{name}</div>
        <div data-testid="test-description">{description}</div>
        <div>
          {member
            ? (
              <span className="mission-status-Active">Active Member</span>)
            : <span className="mission-status">NOT A MEMBER</span> }
        </div>
        <div>
          {member
            ? (
              <button type="button" className="leaveMission" onClick={() => dispatch(joinMission(id))}>
                leave Mission
              </button>
            )
            : (
              <button type="button" className="joinMission" onClick={() => dispatch(joinMission(id))}>
                Join Mission
              </button>
            )}
        </div>
      </div>
    </div>
  );
};
Mission.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  member: PropTypes.bool.isRequired,
  // member: PropTypes.string.isRequired,
};
export default Mission;
