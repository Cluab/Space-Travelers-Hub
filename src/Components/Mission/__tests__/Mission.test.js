import { render, screen } from '@testing-library/react';
// import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import Mission from '../Mission';
import '@testing-library/jest-dom/extend-expect';
import MissionsReducer from '../../../Redux/Missions/mission';

const mission = [{
  id: '9D1B7E0',
  name: 'Thaicom',
  description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
  member: 'false',
}];
const store = configureStore({ reducer: { missions: MissionsReducer } });

describe('nav bar test cases', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Mission
            key={mission[0].id}
            id={mission[0].id}
            name={mission[0].name}
            description={mission[0].description}
            member={mission[0].member}
          />
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match with input data ', () => {
    render(
      <Provider store={store}>
        <Mission
          key={mission[0].id}
          id={mission[0].id}
          name={mission[0].name}
          description={mission[0].description}
          member={mission[0].member}
        />
      </Provider>,
    );
    const missionName = screen.getByTestId('test-name').textContent;
    const missionDescription = screen.getByTestId('test-description').textContent;

    expect(missionName).toBe(mission[0].name);
    expect(missionDescription).toBe(mission[0].description);
  });
});
