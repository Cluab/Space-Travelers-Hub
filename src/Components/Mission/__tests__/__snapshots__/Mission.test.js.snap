// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav bar test cases Link renders correctly 1`] = `
<div
  className="mission-container"
>
  <div
    className="mission-info"
  >
    <div
      data-testid="test-name"
    >
      Thaicom
    </div>
    <div
      data-testid="test-description"
    >
      Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
    </div>
    <div>
      <span
        className="mission-status-Active"
      >
        Active Member
      </span>
    </div>
    <div>
      <button
        className="leaveMission"
        onClick={[Function]}
        type="button"
      >
        leave Mission
      </button>
    </div>
  </div>
</div>
`;
